### PR TITLE
Show commit in `--version` and ledger-tool's log

### PR DESCRIPTION
**Pull request by ryoqun**
#### Problem

we miss commit hash in versions

#### Summary of Changes

Show it. Also, some other niceties I've buffered. ;)

Fixes https://github.com/solana-labs/solana/pull/12376/files#r498626855

### CODE REVIEW

**Comment by ryoqun** [comment]
Sample output

```
[2020-10-02T08:05:52.612554400Z INFO  solana_validator] solana-validator 1.4.0 (src:0562c961; feat:1826818093)
[2020-10-02T08:05:52.612578480Z INFO  solana_validator] Starting validator with: ArgsOs {
```

```
$ ./target/release/solana-ledger-tool --ledger config/ledger verify
[2020-10-02T08:05:39.442722835Z INFO  solana_ledger_tool] solana-ledger-tool 1.4.0 (src:0562c961; feat:1826818093)
genesis hash: GfjReF6EgWD842cup99YsBwyiJt1N5FVWB4DU6BtcUHq
[2020-10-02T08:05:39.445147747Z INFO  solana_ledger::blockstore] Maximum open file descriptors: 500000
```

```
$ ./target/release/solana-validator --version
solana-validator 1.4.0 (src:0562c961; feat:1826818093)
```

File: version/src/lib.rs [hunk 1]
```diff
@@ -91,12 +91,19 @@ impl fmt::Debug for Version {
 impl Sanitize for Version {}
 
 #[macro_export]
-macro_rules! version {
+macro_rules! short_version {
```
Comment (mvines) [inline version/src/lib.rs:94]: Oh how about calling this `solana_version::semver!()`

File: ledger-tool/src/main.rs [hunk 2]
```diff
@@ -1246,6 +1246,8 @@ fn main() {
         )
         .get_matches();
 
+    info!("{} {}", crate_name!(), solana_version::version!());
```
Comment (mvines) [inline ledger-tool/src/main.rs:1249]: Do we need this log?

Comment (ryoqun) [reply ledger-tool/src/main.rs:1249]: @mvines Yeah, this is useful for developing/bisecting. When I finds out that something bad started to occur, I want to know the exact commit hash, which started the odd behavior. I tend to `(./target/release/solana-ledger-tool --version && ./target/release/solana-ledger-tool ...) &> /path/to/log`. And it's tedious at least. Also, `solana-validator` does already this, why not for `solana-ledger-tool`?

Comment (mvines) [reply ledger-tool/src/main.rs:1249]: Got it, thanks for the explanation

**Review by mvines — APPROVED** [review]

**Pull request merged**